reV/bespoke/place_turbines.py

```diff
@@ -0,0 +1,234 @@
+# -*- coding: utf-8 -*-
+"""
+place turbines for bespoke wind plants
+"""
+import numpy as np
+from shapely.geometry import Polygon, MultiPolygon
+import shapely.affinity
+import rasterio.features
+
+from reV.bespoke.pack_turbs import PackTurbines
+from reV.bespoke.gradient_free import GeneticAlgorithm
+
+
+class PlaceTurbines():
+    """Framework for optimizing turbine locations for site specific
+    exclusions, wind resources, and objective
+    """
+    def __init__(self, wind_plant, objective_function, cost_function,
+                 include_mask, pixel_side_length, min_spacing, ga_time):
+        """
+        Parameters
+        ----------
+        wind_plant : WindPowerPD
+            wind plant object to analyze wind plant performance. This object
+            should have everything in the plant defined, such that only the
+            turbine coordinates and plant capacity need to be defined during
+            the optimization.
+        objective_function : str
+            The objective function of the optimization as a string, should
+            return the objective to be minimized during layout optimization.
+            Variables available are:
+                - n_turbines: the number of turbines
+                - system_capacity: wind plant capacity
+                - aep: annual energy production
+                - self.wind_plant: the SAM wind plant object, through which
+                all SAM variables can be accessed
+                - cost: the annual cost of the wind plant (from cost_function)
+        cost_function : str
+            The cost function as a string, should return the annual cost
+            of the wind farm. Variables available are:
+                - n_turbines: the number of turbines
+                - system_capacity: wind plant capacity
+                - aep: annual energy production
+                - self.wind_plant: the SAM wind plant object, through which
+                all SAM variables can be accessed
```

Comment:
Great docstrings. Clever that self.wind_plant is accessible to access any other SAM variables.